live-build/build.gradle

```diff
@@ -75,13 +122,72 @@ for (variant in allVariants) {
             }
 
             outputs.file "${buildDir}/artifacts/${variant}-${platform}.debs.tar.gz"
+
+            commandLine "${rootProject.projectDir}/scripts/run-live-build.sh", variant, platform, upgradeArtifactsRunType
+        }
+
+        tasks.create("clean${taskSuffix}${upgradeArtifactsTaskSuffix}", Exec) {
+            commandLine "rm", "-rf", "${buildDir}/${variant}-${platform}"
+        }
+        tasks.clean.dependsOn "clean${taskSuffix}${upgradeArtifactsTaskSuffix}"
+    }
+}
+
+/*
+ * Create tasks to build VM artifacts. For instance, the task 'buildInternalDevEsxVmArtifacts' will produce
+ * an artifact 'internal-dev-esx.ova'. These tasks depend on their configuration task counterpart.
+ */
+for (variant in allVariants) {
+    for (platform in allPlatforms) {
+        def taskSuffix = "${toCamelCase(variant).capitalize()}${platform.capitalize()}"
```

Comment:
As discussed offline, I've moved most of the work into `run-live-build.sh` instead of gradle tasks so that incremental builds won't break (updated PR description)
Also parameterized the run type into a for loop to remove duplicated code.